Sliders.py

```diff
@@ -0,0 +1,302 @@
+        layout = QGridLayout(self._main)
+
+        # Create a FigureCanvas
+        self.canvas = FigureCanvas(self.figure)
+
+        # Add the canvas to the layout
+        height_plot = 1
+        width_plot = 6
+        layout.addWidget(self.canvas,0,0,height_plot,width_plot)
+        self.addToolBar(NavigationToolbar(self.canvas, self))
+
+        # Offset to make space for the log/linear buttons 
+        shift = 2
+
+        # Buttons for setting plot scale
+        self.lin_button = QRadioButton("Linear")
+        self.log_button = QRadioButton("Log")
+
+        self.lin_button.toggled.connect(lambda: self.change_scale('linear'))
+        self.log_button.toggled.connect(lambda: self.change_scale('log'))
+
+        layout.addWidget(self.lin_button,1,4)
+        layout.addWidget(self.log_button,1,5)
+
+        for dim in self.dims:
+
+            # Create a label
+            dim.create_label()
+
+            # Create X and Y buttons for the slider
+            x_func = self.press_button(dim,0,1)
+            y_func = self.press_button(dim,1,0)
+            dim.create_buttons(x_func,y_func)
+
+            # Create a slider
+            dim.create_slider(self.slider_changer_creator(dim))
+
+            # Create a stepper
+            dim.create_stepper(self.stepper_changer_creator(dim))
+
+            # Add the buttons and slider to the box
+            layout.addWidget(dim.label, dim.no+shift, 0)
+            layout.addWidget(dim.buttons[0], dim.no+shift, 1)
+            layout.addWidget(dim.buttons[1], dim.no+shift, 2)
+            layout.addWidget(dim.slider, dim.no+shift, 3)
+            # Have the slider take up two 'cells' so the log/linear buttons are not pushed too far apart
+            layout.addWidget(dim.stepper, dim.no+shift, 4,1,2)
+
+        # Prepare the initial view (last two dimensions are set to X and Y)
+        self.prepare_initial_view()
+
+    def prepare_initial_view(self):
+
+	# Use the first value of the first n - 2 dimensions for the as the initial slices
+        for i in range(self.n_dims - 2):
+            self.slice_selection[self.dim_names[i]] = 0
+
+        # Set the last two dimensions as the x and y axes
+        self.axes = [self.dims[-2], self.dims[-1]]
+
+        # Reshape the array for the initial configuration
+        self.arr = self.xarr.isel(self.slice_selection).transpose(self.dim_names[-1],self.dim_names[-2])
+
+        # Check the X and Y buttons for the initial axes setup
+        self.axes[0].buttons[0].setChecked(True)
+        self.axes[1].buttons[1].setChecked(True)
+
+        # Disable their sliders and steppers
+        for dim in self.axes:
+
+            dim.slider.setVisible(False)
+            dim.stepper.setVisible(False)
+
+        # Plot the array
+        self.im = self.ax.imshow(self.arr)
+        self.cbar = self.figure.colorbar(self.im)
+
+        # Label the axes
+        self.ax.set_xlabel(self.axes[0].name)
+        self.ax.set_ylabel(self.axes[1].name)
```

Comment:
A lot of code in this function is the same as `change_view`. See if you can just call `change_view` instead of duplicating the code.

Reply:
My `prepare_initial_view` function creates the AxesImage and Colorbar objects while my `change_view` function assumes these objects already exists and just updates them. The initial plot doesn't require an "update" so I instead decided to take some of the shared code and put it in its own functions. These functions are then called by both `prepare_initial_view` and `change_view`. Do you consider this appropriate?

Reply:
Yes, that sounds more or less like what I had in mind.